### PR TITLE
💎💎GEMini updated

### LINKED ISSUE #8699: Gemini Handler Updates
### Short description and motivation for the proposed feature

Add following features to gemini:
- vision mode
- prompt template mode
- embedding mode
- question with context
- question without context
- json struct

### Video or screenshots

_No response_

### Describe some possible solutions

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by parthiv11**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #8699 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

https://github.com/mindsdb/mindsdb/assets/75653580/fe1af932-dcc5-4819-9d7c-912ab8bb8f1a

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by torrmal** [comment]
amazing @paxcema what do you think?

**Comment by parthiv11** [comment]
https://github.com/mindsdb/mindsdb/assets/75653580/bfd1c614-ffa4-438a-9faa-6711d3ea41df

**Comment by parthiv11** [comment]
@paxcema now its support's Json extraction also

**Review by paxcema — CHANGES_REQUESTED** [review]
Generally looks good. However, there is some code duplication from the OpenAI handler. That's alright for the sake of updating this integration, but could you please:

> add a disclaimer at the top of every duplicated code so that we know to refactor it, and open an issue so that we can work on it at a later stage


**Ideally**, you'd be able to follow-up on this, either on this PR or on a subsequent one.

Refactoring any duplicated snippets into common utilities for all handlers to leverage is an important task we have, to keep the codebase healthy. It is also nice to see that common patterns do emerge from LLM integrations 😄

File: mindsdb/integrations/handlers/google_gemini_handler/requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
-google-generativeai==0.3.1
\ No newline at end of file
+google-generativeai==0.3.2
+Pillow==10.1.0
```
[the remark this answers is not in the document]
Comment (parthiv11) [reply mindsdb/integrations/handlers/google_gemini_handler/requirements.txt:2]: no i think it not necessary, therefore i am removing it

**Comment by parthiv11** [comment]
@paxcema sorry for being so late, please check updated code.

**Comment by parthiv11** [comment]
i can open a new PR for refactor, if it sounds good to you

**Comment by parthiv11** [comment]
@paxcema you can merge this PR

**Comment by paxcema** [comment]
@parthiv11 copying comment from #8861 here, for reference. Sorry about the delay:

> we'll kindly ask that you please work on this one in the same branch and pull request (https://github.com/mindsdb/mindsdb/pull/8701), do let us know when it's ready for review, thanks for your contribution!!

In other words, please add the refactor to this PR. We'd rather merge this integration from a single PR that contains all needed for great quality and UX.

**Comment by parthiv11** [comment]
@paxcema ??

**Review by paxcema — APPROVED** [review]
Changes look OK. Tagging @ZoranPandovski for visibility.

@parthiv11 please merge `main` into this, and I will get the workflows running to check tests in the CI. Once that is done, we can merge. Thanks for your patience!

**Comment by parthiv11** [comment]
thank you @paxcema 
@ZoranPandovski  done

**Comment by paxcema** [comment]
@parthiv11 static code check is failing, try `flake8 ./` in your handler folder to spot all the relevant lines that must be fixed

**Comment by parthiv11** [comment]
@paxcema and @ZoranPandovski I check workflows and it worked ed, please verify it

**Comment by parthiv11** [comment]
@ZoranPandovski  i think is PR is done, please checkout it

**Pull request merged**